3.5 branch doesn't compile on Solaris 10

Hi,

I'm seeing an issue compiling syslog-ng 3.5.6 on Solaris 10, with a missing getline function:

Undefined                       first referenced
 symbol                             in file
getline                             ./lib/.libs/libsyslog-ng.so

Where getline is called in lib/filter/filter-in-list.c:

  CC       lib/filter/lib_libsyslog_ng_la-filter-in-list.lo
lib/filter/filter-in-list.c: In function 'filter_in_list_new':
lib/filter/filter-in-list.c:85:3: warning: implicit declaration of function 'getline' [-Wimplicit-function-declaration]
   while (getline(&line, &n, stream) != -1)

As I understand it, Solaris 10's libc is missing this fuction. This was bug 275 in the old BalaBit bugzilla, where it closed, but seemingly without resolution.

https://bugzilla.balabit.com/show_bug.cgi?id=275

Was able to use the user-supplied patch from the Bugzilla  report to misc.c and misc.h to get a build going.

Sun STREAMS module compiles fine still. ^_^

Thanks,

Simon

--- a/lib/filter/filter-in-list.c
+++ b/lib/filter/filter-in-list.c
@@ -1,6 +1,6 @@
 /*
- * Copyright (c) 2013 BalaBit IT Ltd, Budapest, Hungary
- * Copyright (c) 2013 Gergely Nagy <algernon@balabit.hu>
+ * Copyright (c) 2013, 2014 BalaBit IT Ltd, Budapest, Hungary
+ * Copyright (c) 2013, 2014 Gergely Nagy <algernon@balabit.hu>
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -64,8 +64,7 @@ filter_in_list_new(const gchar *list_file, const gchar *property)
 {
   FilterInList *self;
   FILE *stream;
-  size_t n;
-  gchar *line = NULL;
+  gchar line[16384];
 
   stream = fopen(list_file, "r");
   if (!stream)
@@ -82,12 +81,11 @@ filter_in_list_new(const gchar *list_file, const gchar *property)
   self->value_handle = log_msg_get_value_handle(property);
   self->tree = g_tree_new((GCompareFunc) strcmp);
 
-  while (getline(&line, &n, stream) != -1)
+  while (fgets(line, sizeof(line), stream) != NULL)
     {
       line[strlen(line) - 1] = '\0';
       if (line[0])
         g_tree_insert(self->tree, line, GINT_TO_POINTER(1));
-      line = NULL;
     }
   fclose(stream);
 

